js/src/forum/index.ts

```diff
@@ -9,25 +9,25 @@ import ItemList from 'flarum/common/utils/ItemList';
 import { VnodeDOM } from 'Mithril';
 
 app.initializers.add('flarum-pusher', () => {
-  app.pusher = new Promise((resolve) => {
-    $.getScript('//cdn.jsdelivr.net/npm/pusher-js@7.0.3/dist/web/pusher.min.js', () => {
-      const socket: PusherTypes.default = new Pusher(app.forum.attribute('pusherKey'), {
-        authEndpoint: `${app.forum.attribute('apiUrl')}/pusher/auth`,
-        cluster: app.forum.attribute('pusherCluster'),
-        auth: {
-          headers: {
-            'X-CSRF-Token': app.session.csrfToken,
-          },
+  app.pusher = new Promise(async (resolve) => {
```

Comment:
Using `new Promise()` and `async () => {}` is an anti-pattern. We should use either one or the other. (I'd prefer an async function.)

Reply:
@davwheat I'm probably not thinking right at this late hour, but something like this doesn't work:
```ts
  app.pusher = async () => {
    await import('//cdn.jsdelivr.net/npm/pusher-js@7.0.3/dist/web/pusher.min.js' /* webpackIgnore: true, webpackPrefetch: true */);

    const socket: PusherTypes.default = new Pusher(app.forum.attribute('pusherKey'), {
      authEndpoint: `${app.forum.attribute('apiUrl')}/pusher/auth`,
      cluster: app.forum.attribute('pusherCluster'),
      auth: {
        headers: {
          'X-CSRF-Token': app.session.csrfToken,
        },
      },
    });

    return {
      channels: {
        main: socket.subscribe('public'),
        user: app.session.user ? socket.subscribe(`private-user${app.session.user.id()}`) : null,
      },
      pusher: socket,
    };
  };
```

Reply:
Ah sorry, my typo.

It can be simplified to:

```ts
const a = (async () => {
  return await someAsyncCode();
})()
```